optimum/habana/diffusers/pipelines/stable_diffusion_xl/pipeline_stable_diffusion_xl_inpaint.py

```diff
@@ -748,8 +748,6 @@ def denoising_value_valid(dnv):
                 ).to(device=device, dtype=latents.dtype)
 
             self._num_timesteps = len(timesteps)
-            if hasattr(self.scheduler, "set_begin_index"):
-                self.scheduler.set_begin_index()
```

Comment:
@dsocek I don't remember why we added these two lines, do you?

Reply:
@regisss yes I remember.  We added support to define different number of batches to run inference with (good feature for properly assessing performance after warmup) and also we let users pick different schedulers.  To make sure we properly "reset" the selected scheduler after each batch is processed we use `init_step_index` built-in function (in the past we modified the scheduler ourselves but later find this to be supported by diffusers anyway).   However, the `init_step_index` function only works properly if we explicitly set the initial scheduler index value with  
`set_begin_index`.  Hence, we added `set_begin_index` which worked in all test cases as expected.